config/src/main/kotlin/org/springframework/security/config/web/servlet/AuthorizeRequestsDsl.kt

```diff
@@ -152,9 +200,13 @@ class AuthorizeRequestsDsl : AbstractRequestMatcherDsl() {
                     is MatcherAuthorizationRule -> requests.requestMatchers(rule.matcher).access(rule.rule)
                     is PatternAuthorizationRule -> {
                         when (rule.patternType) {
-                            PatternType.ANT -> requests.antMatchers(rule.pattern).access(rule.rule)
+                            PatternType.ANT -> requests.antMatchers(rule.httpMethod, rule.pattern).access(rule.rule)
```

Comment:
We should only use `.antMatchers(HttpMethod, String)` if `httpMethod` is not null, similar to `mvcMatchers`

Reply:
I can do this, although I notice there is a discrepency between the docs for `mvcMatchers()` and `antMatchers()`.

The [documentation for `antMatchers(HttpMethod httpMethod, String... antPatterns)`](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.html#antMatchers-org.springframework.http.HttpMethod-java.lang.String...-) says:

> `method` - the `HttpMethod` to use or `null` for any HttpMethod.

And indeed, the non-`HttpMethod` version [just calls the HttpMethod version with `null`](https://github.com/spring-projects/spring-security/blob/b93528138e2b2f7ad34d10a040e6b7ac50fc587a/config/src/main/java/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.java#L286-L288).

However, the [documentation for `mvcMatchers(HttpMethod method, String... mvcPatterns)`](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.html#mvcMatchers-org.springframework.http.HttpMethod-java.lang.String...-) does not contain the "or `null`" option for the `method` argument (maybe it should?):

> method - the HTTP method to match on

Although, `mvcMatchers(String... mvcPatterns)` does [internally use a protected method that documents the same behaviour](https://github.com/spring-projects/spring-security/blob/b93528138e2b2f7ad34d10a040e6b7ac50fc587a/config/src/main/java/org/springframework/security/config/annotation/web/AbstractRequestMatcherRegistry.java#L163-L170).

I think it's cleaner for them both to use the option to pass in `null`, as it is publicly documented behaviour and will remove the need for the conditional check here. But as it's not documented for `mvcMatchers()`, I'm hesitant. What do you think?

I agree they should be unified, so happy to add the conditional check to `antMatchers()` too if you prefer that.

Reply:
I pushed the version without the conditionals for now, so you can see what I mean.

Reply:
Thanks that looks good to me.